### DISCUSSION TITLE
Use wagmi hooks in Nextjs getServerSideProps

### DISCUSSION THREAD

**Discussion by Ajiva-D** [wevm/wagmi#1426] [Q&A]
I'm trying to use getServerSideProps to get some values in Nextjs, is there a way to use wagmi hooks as regular functions and not react hooks since getServerSideProps can't use react hooks?

**Comment by tmm** [comment]
You can import VanillaJS actions from the `wagmi/actions` entrypoint.

```ts
import { fetchEnsName } from 'wagmi/actions'

export async function getServerSideProps(context) {
  const ensName = await fetchEnsName({ address: '0x…' })
  return {
    props: {},
  }
}
```

Check out the [`@wagmi/core` docs](https://wagmi.sh/core) for more info on actions proxied through to `wagmi/actions`.
